### PR TITLE
Added example using the 'with' statement for auto-logout

### LINKED ISSUE #48: Example using ContextManager is helpful
It will be very helpful to give this example how to use redfish with ContextManager :

```
with  redfish.redfish_client('https://in-g0001-bmca', 'admin', 'password') as REDFISH_OBJ:
    response = REDFISH_OBJ.get("/redfish/v1/systems/1")
    print response.text
```

### LINKED ISSUE #49: Example using ContextManager is helpful
It will be very helpful to give this example how to use redfish with ContextManager :

```
with  redfish.redfish_client('https://192.168.1.100', 'admin', 'password') as REDFISH_OBJ:
    response = REDFISH_OBJ.get("/redfish/v1/systems/1")
    print response.text
```

It will save login and with suto-logout in case of error... No need in additional exception handling with `finally`

### PR DESCRIPTION
**Pull request by mraineri**
Fix #48 
Fix #49

### CODE REVIEW

File: examples/context_manager.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# When running remotely connect using the address, account name,
+# and password to send https requests
+login_host = "https://192.168.1.100"
+login_account = "admin"
+login_password = "password"
+
+## Create a REDFISH object
+with redfish.redfish_client(base_url=login_host, username=login_account, password=login_password) as REDFISH_OBJ:
+    # Login into the server and create a session
+    REDFISH_OBJ.login(auth="session")
```
Comment (glimchb) [inline examples/context_manager.py]: log is already done on __enter__ so you can skip it

Comment (mraineri) [reply examples/context_manager.py]: True... Hmmm, unfortunate that it defaults to Basic Auth for that... I can see people using sessions more....

Comment (glimchb) [reply examples/context_manager.py]: Maybe better to provide default auth method in c-tor `__init__` 
And login() will use it if not provided... 

Anyhow I'm glad with new example! 👍

Comment (mraineri) [reply examples/context_manager.py]: @Yergidy , do you have any opinion in changing the default authentication scheme to be session based? See https://github.com/DMTF/python-redfish-library/blob/master/src/redfish/rest/v1.py#L877

**Comment by mraineri** [comment]
Approved 6/27; will follow up with another pull request if others agree that we can change the default auth type to session.

**Pull request merged**